### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by Shevtsooov**

### CODE REVIEW

File: src/herbivoresAndCarnivores.js [hunk 1]
```diff
@@ -1,15 +1,37 @@
...
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(beast) {
+    if (beast instanceof Herbivore && !beast.hidden) {
+      beast.health -= 50;
+    }
+
+    if (beast.health <= 0) {
+      Animal.alive = Animal.alive.filter(animal => animal.name !== beast.name);
```
Comment (33Orange) [inline src/herbivoresAndCarnivores.js:32]: Let's move this logic to method in instance of `Animal` class. 
You can simply clear the array in this filter from animals that have health <= 0

```suggestion
      Animal.alive = Animal.alive.filter(animal => animal.health > 0);
```

**Review by 33Orange — CHANGES_REQUESTED** [review]
Good job.

Just a one moment

File: src/herbivoresAndCarnivores.js [hunk 2]
```diff
@@ -1,15 +1,41 @@
...
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  removeAnimal() {
```
Comment (dinaminahi) [inline src/herbivoresAndCarnivores.js:10]:
```suggestion
  removeDeadAnimals() {
```

**Review by dinaminahi — APPROVED** [review]
Good job! 👍

File: src/herbivoresAndCarnivores.js (hunk 2 above)
Comment (Shevtsooov) [reply src/herbivoresAndCarnivores.js:10]: Thanks!

**Pull request closed without merging**